src/main/java/com/couchbase/intellij/tree/cblite/CBLTreeRightClickListener.java

```diff
@@ -149,62 +216,108 @@ public void actionPerformed(@NotNull AnActionEvent e) {
         AnAction createDocument = new AnAction("Create Document") {
             @Override
             public void actionPerformed(@NotNull AnActionEvent e) {
-                CBLOpenDocumentDialog dialog = new CBLOpenDocumentDialog(true, project, tree, userObject.getScope(), userObject.getText());
+                CBLOpenDocumentDialog dialog = new CBLOpenDocumentDialog(true, project, tree, userObject.getScope(),
+                        userObject.getText());
                 dialog.show();
             }
         };
         actionGroup.add(createDocument);
 
+        AnAction deleteCollection = new AnAction("Delete Collection") {
+            @Override
+            public void actionPerformed(@NotNull AnActionEvent e) {
+                int result = Messages.showYesNoDialog("<html>Are you sure you want to delete the collection <strong>"
+                        + userObject.getText() + "</strong>?</html>", "Delete Collection", Messages.getQuestionIcon());
+                if (result != Messages.YES) {
+                    return;
+                }
+
+                try {
+
+                    if (ActiveCBLDatabase.getInstance() != null
+                            && ActiveCBLDatabase.getInstance().getDatabase() != null) {
+                        ActiveCBLDatabase.getInstance().getDatabase().deleteCollection(userObject.getText(),
+                                userObject.getScope());
+                        DefaultMutableTreeNode parentNode = (DefaultMutableTreeNode) clickedNode.getParent();
+                        if (parentNode != null && tree.getModel() instanceof DefaultTreeModel) {
```

Comment:
I won't even review this code because a lot of checks here are looking really awkward. Feels like a code that was autogenerated. Refactor your code first to something that looks understandable.

Reply:
Have removed all the extra null checks